process/peer/process.go

```diff
@@ -591,7 +591,7 @@ func (vs *validatorStatistics) ProcessRatingsEndOfEpoch(
 					continue
 				}
 			} else {
-				if validator.List != string(core.EligibleList) && !validatorInfo.IsLeavingEligible(validator) {
+				if validator.List != string(core.EligibleList) && !validatorInfo.WasLeavingEligibleInCurrentEpoch(validator) {
```

Comment:
WasJailedEligibleInCurrentEpoch should be taken into consideration? if yes, i think the condition should be
if !vInfo.WasEligibleInCurrentEpoch(validatorInfo)

Reply:
+1 for Sebi's idea. one function

Reply:
the condition is required as we want to process the rating change for the validators that were active either in the Eligible list, or in the Leaving list (did unstake but while active in the last epoch). 
@SebastianMarian The reason why WasEligibleInCurrentEpoch is not used, is because there is no reason to compute for jailed nodes (that were still active), as anyway for them the rating change has no effect. They still need to pay for the unjail which will set their rating to 50, no matter what it was before.